Brokerages/Bitfinex/BitfinexBrokerage.Util.cs

```diff
@@ -0,0 +1,164 @@
+            { new OrderTypeMap { BitfinexOrderType = "exchange limit", Wallet = WalletType.exchange.ToString(), OrderType = OrderType.Limit } },
+            { new OrderTypeMap { BitfinexOrderType = "exchange stop", Wallet = WalletType.exchange.ToString(), OrderType = OrderType.StopMarket } },
+            //{ new OrderTypeMap { BitfinexOrderType = "exchange trailing stop", Wallet = WalletType.exchange.ToString(), OrderType = OrderType.StopLimit } },
+
+            { new OrderTypeMap { BitfinexOrderType = "market", Wallet = WalletType.trading.ToString(), OrderType = OrderType.Market } },
+            { new OrderTypeMap { BitfinexOrderType = "limit", Wallet = WalletType.trading.ToString(), OrderType = OrderType.Limit } },
+            { new OrderTypeMap { BitfinexOrderType = "stop", Wallet = WalletType.trading.ToString(), OrderType = OrderType.StopMarket } },
+            //{ new OrderTypeMap { BitfinexOrderType = "trailing stop", Wallet = WalletType.trading.ToString(), OrderType = OrderType.StopLimit } },
+        };
+        #endregion
+
+
+        /// <summary>
+        /// Map exchange status
+        /// </summary>
+        /// <param name="response"></param>
+        /// <returns></returns>
+        public static OrderStatus MapOrderStatus(BitfinexOrderStatusResponse response)
+        {
+            decimal remainingAmount;
+            decimal executedAmount;
+            if (response.IsCancelled)
+            {
+                return OrderStatus.Canceled;
+            }
+            else if (decimal.TryParse(response.RemainingAmount, out remainingAmount) && remainingAmount > 0
+                && decimal.TryParse(response.ExecutedAmount, out executedAmount) && executedAmount > 0)
+            {
+                return OrderStatus.PartiallyFilled;
+            }
+            else if (response.IsLive)
+            {
+                return OrderStatus.Submitted;
+            }
+
+            return OrderStatus.Invalid;
+        }
+
+
+        /// <summary>
+        /// Map exchange order type
+        /// </summary>
+        /// <param name="orderType"></param>
+        /// <returns></returns>
+        public string MapOrderType(OrderType orderType)
+        {
+            var result = _orderTypeMap.Where(o => o.Wallet == Wallet && o.OrderType == orderType);
+
+            if (result != null & result.Count() == 1)
+            {
+                return result.Single().BitfinexOrderType;
+            }
+
+            throw new Exception("Order type not supported: " + orderType.ToString());
+        }
+
+
+        /// <summary>
+        /// Map exchange order type
+        /// </summary>
+        /// <param name="orderType"></param>
+        /// <returns></returns>
+        public OrderType MapOrderType(string orderType)
+        {
+            var result = _orderTypeMap.Where(o => o.Wallet == Wallet && o.BitfinexOrderType == orderType);
+
+            if (result != null & result.Count() == 1)
```

Comment:
The single `&` is a non short circuiting operator and as such the null check would not avoid the `.Count()` call.
It should be replaced with the normal `&&` operator.